src/test/itunesxmlimportertest.cpp

```diff
@@ -0,0 +1,339 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <QDir>
+#include <QString>
+#include <atomic>
+#include <memory>
+
+#include "library/itunes/itunesdao.h"
+#include "library/itunes/itunesimporter.h"
+#include "library/itunes/itunesxmlimporter.h"
+#include "test/mixxxtest.h"
+
+class ITunesXMLImporterTest : public MixxxTest {
+  protected:
+    QDir getITunesTestDir() {
+        return MixxxTest::getOrInitTestDir().filePath("itunes/");
+    }
+
+    std::unique_ptr<ITunesXMLImporter> makeImporter(
+            const QString& xmlFileName, std::unique_ptr<ITunesDAO> dao) {
+        QString xmlFilePath = getITunesTestDir().absoluteFilePath(xmlFileName);
+        auto importer = std::make_unique<ITunesXMLImporter>(
+                nullptr, xmlFilePath, cancelImport, std::move(dao));
+        return importer;
+    }
+
+  private:
+    std::atomic<bool> cancelImport;
+};
+
+class MockITunesDAO : public ITunesDAO {
+  public:
+    MOCK_METHOD(bool, importTrack, (const ITunesTrack&));
+    MOCK_METHOD(bool, importPlaylist, (const ITunesPlaylist&));
+    MOCK_METHOD(bool, importPlaylistRelation, (int, int));
+    MOCK_METHOD(bool, importPlaylistTrack, (int, int, int));
+    MOCK_METHOD(bool, applyPathMapping, (const ITunesPathMapping&));
+};
+
+using testing::_;
+using testing::Return;
+
+namespace {
+std::unique_ptr<MockITunesDAO> makeMockDAO() {
+    std::unique_ptr<MockITunesDAO> dao = std::make_unique<MockITunesDAO>();
+
+    ON_CALL(*dao, importTrack(_)).WillByDefault(Return(true));
+    ON_CALL(*dao, importPlaylist(_)).WillByDefault(Return(true));
+    ON_CALL(*dao, importPlaylistRelation(_, _)).WillByDefault(Return(true));
+    ON_CALL(*dao, applyPathMapping(_)).WillByDefault(Return(true));
+
+    return dao;
+}
+
+} // anonymous namespace
+
+TEST_F(ITunesXMLImporterTest, ParseMacOSMusicXML) {
+    std::unique_ptr<MockITunesDAO> dao = makeMockDAO();
+
+    QString musicRoot = getITunesTestDir().absoluteFilePath(
+            "macOS Music Library/Media.localized/Music");
+    EXPECT_CALL(*dao,
+            importTrack(ITunesTrack{
+                    .id = 479,
+                    .artist = "AC/DC",
+                    .title = "Highway to Hell",
+                    .album = "Highway to Hell",
+                    .albumArtist = "AC/DC",
+                    .genre = "Hard Rock",
+                    .grouping = "",
+                    .year = 1979,
+                    .duration = 208,
+                    .location = musicRoot +
+                            "/AC_DC/Highway to Hell/01 Highway to Hell.m4a",
+                    .rating = 0,
+                    .comment = "",
+                    .trackNumber = 1,
+                    .bpm = 0,
+                    .bitrate = 256,
+            }));
```

Comment:
We have a bit of code duplication here with the iTunes XML test below. I have used the same tracks and playlists in the hope of being able to share some logic with the other test. Unfortunately, however, the indexing is different, also there are some minor differences in automatically generated playlists, therefore I believe we cannot (easily) avoid this. Suggestions welcome.